flows/contact.go

```diff
@@ -76,10 +76,24 @@ func (c *Contact) Name() string { return c.name }
 func (c *Contact) URNs() URNList { return c.urns }
 
 // AddURN adds a new URN to this contact
-func (c *Contact) AddURN(urn urns.URN) {
-	// TODO normalize and check we're not adding duplicates
+func (c *Contact) AddURN(urn urns.URN) bool {
+	if c.HasURN(urn) {
+		return false
```

Comment:
originally had this as error case but then noticed we don't blow up if contact_group_added event tries to add a duplicate group